storage/entities/base_entity.js

```diff
@@ -305,33 +305,43 @@ class BaseEntity {
 	parseFilters(filters) {
 		let filterString = null;
 
-		const parseFilterObject = object =>
-			Object.keys(object)
-				.map(key => this.filters[key])
-				.join(' AND ');
-
 		if (Array.isArray(filters)) {
-			filterString = filters
-				.map(filterObject => parseFilterObject(filterObject))
-				.join(' OR ');
-		} else if (typeof filters === 'object') {
-			filterString = parseFilterObject(filters);
+			filterString = this._parseArrayFilter(filters);
+		} else if (Object.keys(filters).length > 0) {
+			filterString = this._parseFilterObject(filters);
 		}
 
-		let filtersObject = filters;
+		const filtersObject = BaseEntity._filtersToObject(filters);
 
-		if (Array.isArray(filters)) {
-			filtersObject = Object.assign({}, ...filters);
-		}
+		return this._filterClause(filterString, filtersObject);
+	}
 
+	_filterClause(filterString, filtersObject) {
+		let filterClause;
 		if (filterString) {
-			return `WHERE ${this.adapter.parseQueryComponent(
+			filterClause = `WHERE ${this.adapter.parseQueryComponent(
 				filterString,
 				filtersObject
 			)}`;
 		}
 
-		return '';
+		return filterClause || '';
+	}
+
+	static _filtersToObject(filters) {
+		return Array.isArray(filters) ? Object.assign({}, ...filters) : filters;
+	}
+
+	_parseArrayFilter(filters) {
```

Comment:
Use `__` double underscore and also add this to conventions list.

Reply:
In `elements` we devided to use `_` for private function/variable. Probably it's better to keep it consistent across product.

Reply:
I do agree with keeping it consistent so I will keep it with single underscore then.